src/features/fund/components/dispenser-api-logged-out.tsx

```diff
@@ -1,15 +1,27 @@
 import { useCallback } from 'react'
 import { Button } from '@/features/common/components/button'
 import { useAuth0 } from '@auth0/auth0-react'
+import { open } from '@tauri-apps/plugin-shell'
 
 export const dispenserApiLoginButtonLabel = 'log in to fund your account'
 
 export function DispenserApiLoggedOut() {
-  const { loginWithPopup } = useAuth0()
+  const { loginWithRedirect, loginWithPopup } = useAuth0()
 
   const logIn = useCallback(async () => {
-    await loginWithPopup()
-  }, [loginWithPopup])
+    if (window.__TAURI_INTERNALS__) {
+      await loginWithRedirect({
+        appState: {
+          returnTo: '/fund',
```

Comment:
Should this be the same url as logout?

Reply:
turns out setting `appState` isn't needed because. when the user is redirected back after authorising, we redirect them to fund page anyway. I deleted this code.

Reply:
I'm assuming that's because it's the redirectUri we have set on the provider itself.